zephyr/esp_shared/components/driver/rtc_io.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * SPDX-FileCopyrightText: 2019-2021 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include "esp_log.h"
+#include "esp_err.h"
+#include "esp_check.h"
+
+#include <zephyr/kernel.h>
+
+#include "driver/rtc_io.h"
+#include "hal/rtc_io_hal.h"
+
+static const char __attribute__((__unused__)) *RTCIO_TAG = "RTCIO";
+
+static int rtc_spinlock;
+
+#if SOC_RTCIO_INPUT_OUTPUT_SUPPORTED
+
+/*---------------------------------------------------------------
+                        RTC IO
+---------------------------------------------------------------*/
+esp_err_t rtc_gpio_init(gpio_num_t gpio_num)
```

Comment:
Same comments as in DAC driver.  
Can we work out the Macro logging system to meet Zephyr's?

Reply:
Yes, it is possible but requires a wider test scope than this PR has.

Other components rely on the ESP macros logging system and could silently break, IMO, it should be addressed on another ticket.

Reply:
Solved in another PR.